### ISSUE TITLE
DigestCredential needs a new interface that supports verifying credentials when given a prehashed user:realm:password

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2280]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#2280|
|Type|enhancement|
|Created|2006-12-07 19:01:42Z|

This is the way that .htdigest files work to avoid keeping the plaintext password on disk.

<details><summary>Searchable metadata</summary>

```
trac-id__2280 2280
type__enhancement enhancement
reporter__dreid dreid
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1165518102000000 1165518102000000
changetime__1173121027000000 1173121027000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_2"></a>|@dreid commented|
|-|-|

covered by [branches/web2-digest-2104](branches/web2-digest-2104)

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_3"></a>|@dreid set status to `closed`|
|-|-|

(In [19763]) Merge web2-digest-2104-2

Author: dreid
Reviewer: therve, glyph, exarkun
Fixes [#2104](https://github.com/twisted/twisted/issues/2104)
Fixes [#2280](https://github.com/twisted/twisted/issues/2280)

Use a verifiable signed opaque to allow clients to reauthenticate 
without forcing a challenge/response authentication to take place 
for every request.

Also make the DigestedCredentials support checking the response against
a precalculated hash of username:realm:password, like that stored in 
.htdigest files.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
